codegen/src/test/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/parsers/XmlBindingTraitSerializerGeneratorTest.kt

```diff
@@ -0,0 +1,152 @@
+
+        map MyMap {
+            @xmlName("Name")
+            key: String,
+
+            @xmlName("Setting")
+            value: Choice,
+        }
+
+        list SomeList {
+            member: Choice
+        }
+
+
+        structure Top {
+            choice: Choice,
+
+            field: String,
+
+            @xmlAttribute
+            extra: Long,
+
+            @xmlName("prefix:local")
+            renamedWithPrefix: String,
+
+            @xmlFlattened
+            recursive: TopList
+        }
+
+        list TopList {
+            member: Top
+        }
+
+        structure OpInput {
+            @httpPayload
+            payload: Top
+        }
+
+        @http(uri: "/top", method: "POST")
+        operation Op {
+            input: OpInput,
+        }
+    """.asSmithyModel()
+
+    @Test
+    fun `generates valid serializers`() {
+        val model = RecursiveShapeBoxer.transform(
+            OperationNormalizer(baseModel).transformModel(
+                OperationNormalizer.NoBody,
+                OperationNormalizer.NoBody
+            )
+        )
+        val symbolProvider = testSymbolProvider(model)
+        val parserGenerator = XmlBindingTraitSerializerGenerator(testProtocolConfig(model))
+        val operationParser = parserGenerator.payloadSerializer(model.lookup("test#OpInput\$payload"))
+
+        val project = TestWorkspace.testProject(testSymbolProvider(model))
+        project.lib { writer ->
+            writer.unitTest(
+                """
+                 use model::Top;
+                let inp = crate::input::OpInput::builder().payload(
+                    Top::builder()
+                        .field("hello!")
+                        .extra(45)
+                        .recursive(Top::builder().extra(55).build())
+                        .build()
+                ).build().unwrap();
+                let serialized = ${writer.format(operationParser)}(&inp.payload.unwrap()).unwrap();
+                let output = std::str::from_utf8(serialized.bytes().unwrap()).unwrap();
+                assert_eq!(output, "<Top extra=\"45\">hello!</Top>");
+
+            """
+            )
+        }
+        project.withModule(RustModule.default("model", public = true)) {
+            model.lookup<StructureShape>("test#Top").renderWithModelBuilder(model, symbolProvider, it)
+            UnionGenerator(model, symbolProvider, it, model.lookup("test#Choice")).render()
+            val enum = model.lookup<StringShape>("test#FooEnum")
+            EnumGenerator(symbolProvider, it, enum, enum.expectTrait()).render()
+        }
+
+        project.withModule(RustModule.default("input", public = true)) {
+            model.lookup<OperationShape>("test#Op").inputShape(model).renderWithModelBuilder(model, symbolProvider, it)
+        }
+        println("file:///${project.baseDir}/src/xml_ser.rs")
+        try {
+            project.compileAndTest()
+        } finally {
+            // "cargo fmt".runCommand(project.baseDir)
```

Comment:
Is this left in to make debugging easier when the test fails? Just uncomment, rerun, and be able to read the code easier?

Reply:
migrated this into `compileAndTest()` where it should have been all along